compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -329,7 +329,7 @@ class TreePickler(pickler: TastyPickler) {
       pickleName(sym.name)
       pickleParams
       tpt match {
-        case _: Template | _: Hole => pickleTree(tpt)
+        case _: Template | _: Literal | _: Hole => pickleTree(tpt)
```

Comment:
I think the problem is in the typer, not the pickler. `Literal` is not a TypeTree so it shouldn't show up in `tpt`, I would expect a `TypeTree(ConstantType(Constant(123)))` instead /cc @nicolasstucki

Reply:
So is there anything wrong with my initial commit that just ignores `Literal`? Is there really a problem in the typer?

Reply:
well we do need to pickle the type to be able to restore it in unpickling so I don't see how the tpt field could be ignored.

Reply:
Yeah, your first commit fails with

```
[info] Test dotty.tools.dotc.CompilationTests.pickling started
java.lang.AssertionError: assertion failed: illegal modifier tag 111 at Addr(333), end = Addr(331) while compiling tests/pos/i13503.scala
```

Reply:
`TypeTree(ConstantType(Constant(123)))` makes more sense. We should find where this tree is created and try to change it.

Reply:
I wondering if `Ycheck` should have caught this as it is a `TermTree` used as a type tree.